components/switch/test/switch.vdiff.js

```diff
@@ -0,0 +1,50 @@
+import '../switch.js';
+import { expect, fixture, focusElem, hoverElem, html } from '@brightspace-ui/testing';
+
+const offFixture = html`<d2l-switch text="Test Text"></d2l-switch>`;
+const onFixture = html`<d2l-switch text="Test Text" on></d2l-switch>`;
+
+describe('d2l-switch', () => {
+
+	describe('ltr', () => {
+
+		describe('basic', () => {
+			[
+				{ name: 'off', template: offFixture },
+				{ name: 'off-focus', template: offFixture, action: async(elem) => await focusElem(elem) },
+				{ name: 'off-disabled', template: html`<d2l-switch text="Test Text" disabled></d2l-switch>` },
+				{ name: 'off-hover', template: offFixture, action: async(elem) => await hoverElem(elem.shadowRoot.querySelector('.d2l-switch-inner')) },
```

Comment:
This was a little fun -- coincidentally `hoverElem(elem)` hovers right in the middle of the switch, which just happens to be between the toggle and the text in a place that _doesn't_ trigger the hover styles.